From 64a9161dd089ecbce738c97df20023b5c2307f0b Mon Sep 17 00:00:00 2001
From: Eric Mehl <eric@thinkhypothetical.com>
Date: Thu, 10 Mar 2022 13:05:28 -0500
Subject: [PATCH] Fix MSVC warning C4701 (Potentially unitialized local
 variable)

- Warnings were being generated by Boost::Wave :
        include\boost\wave\util\cpp_macromap.hpp(1819) : warning C4701: potentially uninitialized local variable 'is_system' used
        include\boost\wave\util\cpp_macromap.hpp(1819) : warning C4701: potentially uninitialized local variable 'is_quoted_filename' used
        include\boost\wave\util\cpp_macromap.hpp(1819) : warning C4701: potentially uninitialized local variable 'is_system' used
        include\boost\wave\util\cpp_macromap.hpp(1819) : warning C4701: potentially uninitialized local variable 'is_quoted_filename' used
---
 SConstruct | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SConstruct b/SConstruct
index cf5865ae4..786946ed8 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2165,6 +2165,7 @@ if env["WITH_GL"] and doConfigure :
 			# while still using -Werror.
 			"-Wno-format" if env["PLATFORM"] != "win32" else "",
 			"-Wno-strict-aliasing" if env["PLATFORM"] != "win32" else "",
+			"/wd4701" if env["PLATFORM"] == "win32" else "",
 			systemIncludeArgument, "$GLEW_INCLUDE_PATH",
 			systemIncludeArgument, "$OIIO_INCLUDE_PATH",
 		],
-- 
2.33.1.windows.1

